llvm/include/llvm/Object/ELFObjectFile.h

```diff
@@ -1022,6 +1027,40 @@ ELFObjectFile<ELFT>::section_rel_begin(DataRefImpl Sec) const {
   uintptr_t SHT = reinterpret_cast<uintptr_t>((*SectionsOrErr).begin());
   RelData.d.a = (Sec.p - SHT) / EF.getHeader().e_shentsize;
   RelData.d.b = 0;
+  if (reinterpret_cast<const Elf_Shdr *>(Sec.p)->sh_type == ELF::SHT_CREL) {
```

Comment:
Is there scope for a forwards iterator that decodes a CREL when dereferenced or incremented. That could be used in a thread-safe environment and wouldn't take up a lot of memory.

May need to be under a different interface.